### DISCUSSION TITLE
HTTP 400 Error with PnP - Search Results

### DISCUSSION THREAD

**Discussion by egwrk** [microsoft-search/pnp-modern-search#3777] [Q&A] [answered]
I have PnP web parts on a SharePoint page that I helped set up for another team over a year ago. They will sometimes have the HTTP 400 error (please see image below) when they go to the page. There are other PnP web parts that were set up on different pages for different purposes in our tenant. However, this particular one seems to have this error much more than the others. Also, when I visit the page in the screenshot below, I do not get the error so not sure if it's an issue with their internet connection or something else. There are currently 106 items that the results should be displaying so the list is not that large either.

I'm not familiar with HTTP errors but is there any way to prevent this error from happening? They have reached out to me about this error on different occasions since the web parts were set up and I'm not sure how to resolve this error. The user experiencing this error has also tried clearing their cache and cookies from the last 4 weeks and they are still getting the error. 

![image](https://github.com/microsoft-search/pnp-modern-search/assets/161059375/ba8bc39f-9503-42c7-96fe-bccfb50e3a7a)

**Comment by wobba** [comment]
400 is a bad request so the question is if something in the webpart setup causing to send a badly formatted request for some scenarios? If it's intermittent and works on refresh, then it's an internal API issue. Looking at the search request JSON may help.

**Comment by egwrk** [comment, marked as answer]
Thanks so much for your response wobba! Not sure why, but when the results were sorted by Title it would give the 400 error so I removed Title from the sort and that seemed to fix it. When sorting with a RefinableString it works fine though. The results worked a couple weeks ago with both Title and RefinableStringX used in the sorting so not sure if something changed.

**Reply by wobba** [reply]
Title is not a sortable managed property. Which would cause this. If you want to sort on Title you have to map to a RefinableString property and use that.

**Reply by kasperbolarsen** [reply]
See https://www.m365thinking.com/post/fix-one-of-the-problems-in-sharepoint-search-make-title-sortable

**Comment by calebmil** [comment]
I too am experiencing this issue.  However, when I tried to rebuild the search results configuration on my page, neither the web part throwing the error nor the replacement web part will allow me to see the list of selected properties.  When I look at the console through developer tools, I see the 400 error coming from an attempt to post a query to the search API followed by an uncaught promise as a result.

In a related problem, when I try to add a selected property that I manually entered, the Layout Slots screen does not recognize the fields I entered (they do not appear in the drop down list of available selected properties).

The page this result web part is on has not been touched in some time but started exhibiting this behavior within the last week.

![image](https://github.com/microsoft-search/pnp-modern-search/assets/6432243/45143900-c9c2-4f3e-9601-1341132b2cf2)

**Reply by kasperbolarsen** [reply]
In most cases where I see that 400 error the root cause is that the Sort settings are wrong. Nothing prevents me from configuring the Sort settings with a property that isn't sortable. However, the API ain't having it, and throws a 400 error. 
Any chance it could be the reason here?

**Reply by calebmil** [reply]
Kasper,

The error appears to be a symptom of a deeper underlying problem.  The field in the sort is a legitimate field that should be in our search index and had been getting retrieved correctly for at least two months.  By removing the sort field, I was able to get the error to go away. 
 
The deeper problem appears to be querying the search API is still returning a 400 error which is preventing the query to retrieve selected properties from the results set from running to completion.  The error I posted above occurs when I am in edit mode and click on Selected Properties.  The progress graphic just spins and never loads the list of available properties.  This still happens even after fixing the sort order problem.  The end result for me is:

- I cannot see selected properties
- Custom properties specific to my result set cannot be used for sorting
- Custom properties specific to my result set are not available for rendering through my template
- When I view the results set in Debug mode, none of my Custom Properties are being displayed.

We're using PnPMSWP 4.9.3.  Is there anything in recent Microsoft releases that would necessitate me upgrading to the latest release of PnPMSWP?

Caleb

**Reply by kasperbolarsen** [reply]
Let's see if we can figure out this one.

Step one could be to eliminate any issues in the current web parts:
So, setting up a basic ( no sorting !) replica of the current search results web part should allow you to determine if the query is sound.
If it works,  try adding the sorting field to Selected Properties and the sort configuration.  If it blows up,  then that sortable property is not sortable 😉
As a rule of thumb we can't assist on any issues if you are not on the most recent version.

**Comment by calebmil** [comment]
I think we are looking at something environmental in our tenant.  This is
occurring with multiple search solutions, not just the one I initially was
called in on.  Time to call MSFT!
